pkg/ttl/cache/table.go

```diff
@@ -287,15 +287,13 @@ func (t *PhysicalTable) SplitScanRanges(ctx context.Context, store kv.Storage, s
 	switch ft.GetType() {
 	case mysql.TypeTiny, mysql.TypeShort, mysql.TypeLong, mysql.TypeLonglong, mysql.TypeInt24:
 		if len(t.KeyColumns) > 1 {
-			return t.splitCommonHandleRanges(ctx, tikvStore, splitCnt, true, mysql.HasUnsignedFlag(ft.GetFlag()))
+			return t.splitCommonHandleRanges(ctx, tikvStore, splitCnt, true, mysql.HasUnsignedFlag(ft.GetFlag()), false)
 		}
 		return t.splitIntRanges(ctx, tikvStore, splitCnt)
 	case mysql.TypeBit:
-		return t.splitCommonHandleRanges(ctx, tikvStore, splitCnt, false, false)
+		return t.splitCommonHandleRanges(ctx, tikvStore, splitCnt, false, false, true)
 	case mysql.TypeString, mysql.TypeVarString, mysql.TypeVarchar:
-		if mysql.HasBinaryFlag(ft.GetFlag()) {
-			return t.splitCommonHandleRanges(ctx, tikvStore, splitCnt, false, false)
-		}
+		return t.splitCommonHandleRanges(ctx, tikvStore, splitCnt, false, false, mysql.HasBinaryFlag(ft.GetFlag()))
```

Comment:
Should it only accept `utf8mb4_bin` (or other binary collations)?

For `utf8mb4_unicode_ci` (or other UCA collations), there is no direct connection between the character and the weight. It's meaningless to get the ASCII prefix.

Reply:
For example, the string `a` has weight `0x0E33`.

Reply:
Done, PTAL again